Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -508,6 +509,87 @@ public sealed class Server : ServerBase, IDisposable {
             }).Concat(extras).ToArray();
         }
 
+        public override async Task<Hover> Hover(TextDocumentPositionParams @params) {
+            var uri = @params.textDocument.uri;
+            GetAnalysis(@params.textDocument, @params.position, @params._version, out var entry, out var tree);
+
+            TraceMessage($"Hover in {uri} at {@params.position}");
+
+            var analysis = entry?.Analysis;
+            if (analysis == null) {
+                TraceMessage($"No analysis found for {uri}");
+                return default(Hover);
+            }
+
+            int? version = null;
+            var parse = entry.WaitForCurrentParse(_clientCaps?.python?.completionsTimeout ?? -1);
+            if (parse != null) {
+                tree = parse.Tree ?? tree;
+                if (parse.Cookie is VersionCookie vc) {
+                    if (vc.Versions.TryGetValue(GetPart(uri), out var bv)) {
+                        tree = bv.Ast ?? tree;
+                        if (bv.Version >= 0) {
+                            version = bv.Version;
+                        }
+                    }
+                }
+            }
+
+            int index = tree.LocationToIndex(@params.position);
+            var w = new ImportedModuleNameWalker(entry.ModuleName, index);
+            tree.Walk(w);
+            ModuleReference modRef;
+            if (!string.IsNullOrEmpty(w.ImportedName) &&
+                _analyzer.Modules.TryImport(w.ImportedName, out modRef)) {
+
+                // Return module information
+                return new Hover { contents = "{0} : module".FormatUI(w.ImportedName) };
+            }
+
+            Expression expr;
+            SourceSpan exprSpan;
+            Analyzer.InterpreterScope scope = null;
+
+            if (!string.IsNullOrEmpty(@params._expr)) {
+                TraceMessage($"Getting hover for {@params._expr}");
```

Comment:
Since we have `TraceMessage(string)`, `TraceMessage(IFormattable)` isn't called. Need to rename one of overloads.

Reply:
I'll just delete the `string` overload and add `$` prefix for all the strings.